src/v/raft/log_eviction_stm.cc

```diff
@@ -16,23 +16,161 @@
 
 namespace raft {
 
+ss::sstring
+log_eviction_stm::eviction_event_type_to_str(eviction_event_type type) {
+    switch (type) {
+    case eviction_event_type::storage:
+        return "storage";
+    case eviction_event_type::kafka:
+        return "kafka";
+    }
+    return "undefined";
+}
+
 log_eviction_stm::log_eviction_stm(
   consensus* raft,
   ss::logger& logger,
   ss::lw_shared_ptr<storage::stm_manager> stm_manager,
   ss::abort_source& as)
-  : _raft(raft)
+  : raft::state_machine(raft, logger, ss::default_priority_class())
```

Comment:
do we want to have default priority class here ? Maybe it should depend from the partition type ?